charts/datadog/templates/_container-agent.yaml

```
@@ -157,10 +157,8 @@
     - name: DD_CHECKS_TAG_CARDINALITY
       value: {{ .Values.datadog.checksCardinality | quote }}
     {{- end }}
-    {{- if .Values.datadog.containerLifecycle.enabled }}
     - name: DD_CONTAINER_LIFECYCLE_ENABLED
-      value: {{ .Values.datadog.containerLifecycle.enabled | quote }}
-    {{- end }}
+      value: {{ .Values.datadog.containerLifecycle.enabled | default true | quote  }}
```

from what I tested, it doesn't work as expected. 

if you set `datadog.containerLifecycle.enabled: false` it still returning "true".
because the `default` function with a boolean value use the default value (in that case `true`) if `false`. 

What I suggest is to juste use `{{ .Values.datadog.containerLifecycle.enabled | quote }}`

or to support an empty ` .Values.datadog.containerLifecycle.enabled` value

```suggestion
      value: {{ .Values.datadog.containerLifecycle.enabled | quote | default "\"true\"" }}
```